### PR TITLE
AQL setup now abort earlier if no shards of a collection can be identified

### PR DESCRIPTION
**Pull request by mchacki**
### Scope & Purpose

We now directly abort if we failed to translate Collection => Shard in AQL setup.
This situation can happen if the collection or database is dropped.
Before we still would return an error and leave no harm, this situation is now just a bit cleaner.

- [ ] :hankey: Bugfix 
- [ ] :pizza: New feature 
- [ ] :hammer: Refactoring 
- [x] :book: CHANGELOG entry made
- [x] :muscle: The behavior in this PR was *manually tested*
- [x] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [ ] No backports required
- [ ] Backports required for: *(Please specify versions)*

#### Related Information

*(Please reference tickets / specification etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket number: https://arangodb.atlassian.net/browse/BTS-158.
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [x] This PR adds tests that were used to verify all changes:
  - [x] Added **Regression Tests**
  - [ ] Added new C++ **Unit Tests**
  - [ ] Added new **integration tests** (i.e. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)

Additionally:

- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Jenkins: http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12472/

### Documentation

> All new Features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the changelog so that
> developers and users have a concise overview. 

- [x] Added a *Changelog Entry* (referencing the corresponding public or internal issue number)
- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by mchacki** [comment]
Jenkins blue, besides known go issue.

File: arangod/Aql/ShardLocking.cpp [hunk 1]
```diff
@@ -152,10 +152,14 @@ void ShardLocking::updateLocking(Collection const* col,
...
-      TRI_ASSERT(false);
+      auto const& name = col->name();
+      if (!TRI_vocbase_t::IsSystemName(name)) {
+        LOG_TOPIC("0997e", WARN, arangodb::Logger::AQL)
+          << "Accessing collection: " << name << " does not translate to any shard. Aborting query."; 
+      }
+      THROW_ARANGO_EXCEPTION_MESSAGE(
+          TRI_ERROR_QUERY_COLLECTION_LOCK_FAILED,
+          "Could not identify any shard belonging to collection: " + name + ". Maybe it is dropped?");
       return;
```
Comment (jsteemann) [inline arangod/Aql/ShardLocking.cpp:163]: We can remove the return statement here now.
```suggestion
```

**Review by jsteemann — APPROVED** [review]
LGTM

File: arangod/Aql/ShardLocking.cpp (hunk 1 above)
Comment (mchacki) [reply arangod/Aql/ShardLocking.cpp:163]: 🤣 you are right, thanks for spotting.

**Pull request merged**